storage_service/src/merkle_tree/hasher.rs

```diff
@@ -1,36 +1,29 @@
-use base64::{engine::general_purpose, Engine as _};
-use ring::digest::{Context, SHA256, SHA384};
+use atlas_common::hash::{calculate_hash, HashAlgorithm};
 use std::fmt::Debug;
 
 /// Trait for hashing functionality
 pub trait Hasher: Send + Sync + Debug {
     fn hash(&self, data: &[u8]) -> String;
 }
 
-/// Default SHA384 hasher implementation
+/// Default SHA384 hasher implementation using atlas-common
 #[derive(Clone, Debug)]
 pub struct DefaultHasher;
 
 impl Hasher for DefaultHasher {
     fn hash(&self, data: &[u8]) -> String {
-        let mut context = Context::new(&SHA384);
-        context.update(data);
-        let digest = context.finish();
-        general_purpose::STANDARD.encode(digest.as_ref())
+        calculate_hash(data) // Uses atlas-common's SHA384 default
     }
 }
 
-/// SHA256 hasher implementation (if needed for compatibility)
+/// SHA256 hasher implementation using atlas-common
 #[derive(Clone, Debug)]
 #[allow(dead_code)]
 pub struct Sha256Hasher;
 
 #[allow(dead_code)]
 impl Hasher for Sha256Hasher {
     fn hash(&self, data: &[u8]) -> String {
-        let mut context = Context::new(&SHA256);
-        context.update(data);
-        let digest = context.finish();
-        general_purpose::STANDARD.encode(digest.as_ref())
+        atlas_common::hash::calculate_hash_with_algorithm(data, &HashAlgorithm::Sha256)
```

Comment:
This function should probably just be included in the import statement at the top.